snake.go

```diff
@@ -73,9 +90,41 @@ func isDigit(c byte) bool {
 	return c >= '0' && c <= '9'
 }
 
-func toLower(c byte) byte {
-	if isUpper(c) {
-		return c + ('a' - 'A')
-	}
-	return c
+var asciiLowercaseArray = [...]byte{
```

Comment:
```suggestion
var asciiLowercaseArray = [256]byte{
```
This will add a second check to ensure that we cover all possible cases, if we were off-by-one the compiler would be inserting bound checks where we access the array.

Reply:
love it! thanks!